assets/js/components/EntitySearchInput.js

```diff
@@ -0,0 +1,56 @@
+/**
+ * EntitySearchInput component.
+ *
+ * Site Kit by Google, Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { useCallback, useState } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import Button from './Button';
+import MagnifyingGlass from '../../svg/magnifying-glass.svg';
+
+const EntitySearchInput = () => {
```

Comment:
Please, use the classic function declaration style for top-level functions and components because there is no sense in using the arrow function for them.
```suggestion
function EntitySearchInput() {
```